pkg/server/datastore/sqldriver/awsrds/awsrds.go

```diff
@@ -85,6 +94,9 @@ func (w *sqlDriverWrapper) Open(name string) (driver.Conn, error) {
 		return nil, fmt.Errorf("could not unmarshal configuration: %w", err)
 	}
 
+	w.tokensMapMtx.Lock()
```

Comment:
I don't know if sqlDriver.Open can block for some time, but if so, maybe we can shorten the scope of this lock so it doesn't block other attempts?

Reply:
It's a good idea to shorten the scope.